src/linear_algebra/bilinear_form.lean

```diff
@@ -1016,3 +1046,274 @@ begin
+  (W : submodule R M) : sym_bilin_form.is_sym $ B.restrict W :=
+λ x y, hB x y
+
+/-- A nondegenerate bilinear form is a bilinear form such that the only element that is orthogonal
+  to every other element is `0`. -/
+def nondegenerate (B : bilin_form R M) : Prop :=
+  ∀ m : M, (∀ n : M, B m n = 0) → m = 0
+
+variables {n : Type w} [fintype n]
+
+/-- A bilinear form is nondegenerate if and only if it has a trivial kernel. -/
+theorem nondegenerate_iff_ker_eq_bot {B : bilin_form R₂ M₂} :
+  B.nondegenerate ↔ B.to_lin.ker = ⊥ :=
+begin
+  rw linear_map.ker_eq_bot',
+  split; intro h,
+  { refine λ m hm, h _ (λ x, _),
+    rw [← to_linear_map_apply, hm], refl },
+  { intros m hm, apply h,
+    ext, exact hm x }
+end
+
+/-- Let `B` be a symmetric, nondegenerate bilinear form on a nontrivial module `M` over the ring
+  `R` with invertible `2`. Then, there exists some `x : M` such that `B x x ≠ 0`. -/
+lemma exists_bilin_form_self_neq_zero [htwo : invertible (2 : R)] [nontrivial M]
+  {B : bilin_form R M} (hB₁ : B.nondegenerate) (hB₂ : sym_bilin_form.is_sym B) :
+  ∃ x, ¬ B.is_ortho x x :=
+begin
+  by_contra, push_neg at h,
+  have : ∀ u v, 2 * B u v = 0,
+  { intros u v,
+    rw [show 2 * B u v = B u u + B v u + B u v + B v v,
+          by rw [is_ortho_def.1 (h u), is_ortho_def.1 (h v), hB₂ v u, two_mul, zero_add]; ring,
+        show B u u + B v u + B u v + B v v = B (u + v) (u + v),
+          by simp [← add_assoc], is_ortho_def.1 (h _)] },
+  have hcon : ∀ u v, B u v = 0,
+  { intros u v,
+    rw [show 0 = htwo.inv_of * (2 * B u v), by rw [this, mul_zero]], simp [← mul_assoc] },
+  exact let ⟨v, hv⟩ := exists_ne (0 : M) in hv $ hB₁ v (hcon v),
+end
+
+variables {V : Type u} {K : Type v}
+variables [field K] [add_comm_group V] [vector_space K V]
+
+/-- A set of orthogonal vectors `v` with respect to some bilinear form `B` is  linearly independent
+  if for all `i`, `B (v i) (v i) ≠ 0`. -/
+lemma linear_independent_of_is_Ortho
+  {n : Type w} {B : bilin_form K V} {v : n → V}
+  (hv₁ : B.is_Ortho v) (hv₂ : ∀ i, ¬ B.is_ortho (v i) (v i)) : linear_independent K v :=
+begin
+  classical,
+  rw linear_independent_iff',
+  intros s w hs i hi,
+  have : B (s.sum $ λ (i : n), w i • v i) (v i) = 0,
+  { rw [hs, zero_left] },
+  have hsum : s.sum (λ (j : n), w j * B (v j) (v i)) =
+    s.sum (λ (j : n), if i = j then w j * B (v j) (v i) else 0),
+  { refine finset.sum_congr rfl (λ j hj, _),
+    by_cases (i = j),
+    { rw [if_pos h] },
+    { rw [if_neg h, is_Ortho_def.1 hv₁ _ _ h, mul_zero] } },
+  simp_rw [map_sum_left, smul_left, hsum, finset.sum_ite_eq] at this,
+  rw [if_pos, mul_eq_zero] at this,
+  cases this,
+  { assumption },
+  { exact false.elim (hv₂ i $ this) },
+  { assumption }
+end
+
+-- ↓ This lemma only applies in fields as we require `a * b = 0 → a = 0 ∨ b = 0`
+lemma span_inf_ortho_eq_bot {B : bilin_form K V} {x : V} (hx : ¬ B.is_ortho x x) :
+  (K ∙ x) ⊓ B.orthogonal (K ∙ x) = ⊥ :=
```

Comment:
Let's rename this, `ortho` could be confused with `is_ortho` so I think `orthogonal` should be written out in full.  Also, you use `span` for this and the next and `singleton` for the one after, but I think `span_singleton` is a fairly set phrase in the library's naming conventions and should also be written in full.

Taking inspiration from the corresponding lemmas for inner product spaces (`submodule.inf_orthogonal_eq_bot`, etc), how about `span_singleton_inf_orthogonal` or `span_singleton_inf_orthogonal_eq_bot` for this one?

The next one could be changed from `span_sup_ortho_eq_top` to `span_singleton_sup_orthogonal`, the one after that to `is_compl_span_singleton_orthogonal`.